eng/pipelines/common/templates/steps/configure-sql-server-win-step.yml

```
@@ -195,6 +195,44 @@ steps:
   displayName: 'Setup SQL Alias [Win]'
   condition: ${{parameters.condition }}
 
+- powershell: |
+    # Create Certificate
+    $computerDnsName = [System.Net.Dns]::Resolve($null).HostName
+    $certificate = New-SelfSignedCertificate -DnsName $computerDnsName,localhost -CertStoreLocation cert:\LocalMachine\My -FriendlyName test99 -KeySpec KeyExchange
+
+    # Get path to Private key (used later)
+    $keyPath = $certificate.PrivateKey.CspKeyContainerInfo.UniqueKeyContainerName
+    $machineKeyPath = "$env:ProgramData\Microsoft\Crypto\RSA\MachineKeys\$keyPath"
+
+    # Add certificate to trusted roots
+    $store = new-object System.Security.Cryptography.X509Certificates.X509Store(
+        [System.Security.Cryptography.X509Certificates.StoreName]::Root,
+        "localmachine"
+    )
+
+    $store.open("MaxAllowed") 
+    $store.add($certificate) 
+    $store.close()
+
+    # Get SQL Server instances and add the Certificate
+    $instances = Get-ItemProperty 'HKLM:\SOFTWARE\Microsoft\Microsoft SQL Server\Instance Names\SQL'
+    foreach ($instance in $instances){
+        $instance | ForEach-Object {
+        $_.PSObject.Properties | Where-Object { $_.Name -notmatch '^PS.*' } | ForEach-Object {
+                Set-ItemProperty "HKLM:\SOFTWARE\Microsoft\Microsoft SQL Server\$($_.Value)\MSSQLServer\SuperSocketNetLib" -Name Certificate -Value $certificate.Thumbprint.ToLower()
+                
+                # Grant read access to Private Key for SQL Service Account
+                if ($($_.Name) -eq "MSSQLSERVER") {
+                    icacls $machineKeyPath /grant "NT Service\MSSQLSERVER:R"
+                } else {
+                    icacls $machineKeyPath /grant "NT Service\MSSQL`$$($_.Name):R"
+                }
```

```suggestion
                $serviceAccount = (Get-ItemProperty "HKLM:\SYSTEM\CurrentControlSet\Services\$($_.Name)" -Name ObjectName).ObjectName
                Set-ItemProperty "HKLM:\SOFTWARE\Microsoft\Microsoft SQL Server\$($_.Value)\MSSQLServer\SuperSocketNetLib" -Name Certificate -Value $certificate.Thumbprint.ToLower()
                
                # Grant read access to Private Key for SQL Service Account
                icacls $machineKeyPath /grant "$serviceAccount:R"
```

This will handle the case where SQL Server runs under a non-default account (CONTOSO\srv-mssql, etc.)

It'd be a little more PowerShell-centric to use the built-in `Get-Acl` and `Set-Acl` rather than shelling out to `icacls`, but that doesn't do anything different - the approach is fine.